### PR TITLE
Remove SCL macros from foreman package & update deps

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Review by evgeni — APPROVED** [review]

**Comment by ekohl** [comment]
Why?
```
[2022-07-15T10:23:59.565Z]     package: foreman-3.4.0-0.5.develop.20220715101833git2641765.el8.noarch from downloaded_rpms
[2022-07-15T10:23:59.565Z]       unresolved deps:
[2022-07-15T10:23:59.565Z]         rubygem(rdoc) < 6.4
```

**Comment by ekohl** [comment]
I took this a step further and removed all SCL macros too.

File: packages/foreman/foreman/foreman.spec [hunk 1]
```diff
@@ -27,40 +22,25 @@ Source4: %{name}.cron.d
...
-
-Obsoletes: foreman-compute < %{version}-%{release}
-Obsoletes: foreman-sqlite < %{version}-%{release}
-Obsoletes: %{?scl_prefix}rubygem-foreman_userdata
-Obsoletes: foreman-rackspace < %{version}-%{release}
-
-Requires: %{?scl_prefix_ruby}ruby(release)
-Requires: %{?scl_prefix_ruby}rubygems
-Requires: %{?scl_prefix_ruby}rubygem(rake) >= 0.8.3
-Requires: %{?scl_prefix_ruby}rubygem(rdoc)
```
Comment (ekohl) [inline packages/foreman/foreman/foreman.spec:41]: Lower we have rdoc < 6.4 so this is redundant.

File: packages/foreman/foreman/foreman.spec [hunk 2]
```diff
@@ -909,18 +864,18 @@ mkdir db/ \\
...
 rm \`pwd\`/config/database.yml \\
-/usr/bin/%%{?scl:%%{scl}-}rake security:generate_encryption_key \\
+%{rake} security:generate_encryption_key \\
 export BUNDLER_EXT_NOSTRICT=1 \\
 export NODE_ENV=production \\
 cp %%{buildroot}%%{%{name}_bundlerd_dir}/%%{gem_name}.rb ./bundler.d/%%{gem_name}.rb \\
-%%{?-s:/usr/bin/%%{?scl:%%{scl}-}rake %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=production DATABASE_URL=nulldb://nohost --trace} \\
-%%{?-a:/usr/bin/%%{?scl:%%{scl}-}rake plugin:apipie:cache[%%{gem_name}] FOREMAN_APIPIE_LANGS=en_US RAILS_ENV=production cache_part=resources OUT=%%{buildroot}%%{%{name}_apipie_cache_plugin} DATABASE_URL=nulldb://nohost --trace} \\
+%%{?-s:%{rake} %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=production DATABASE_URL=nulldb://nohost --trace} \\
+%%{?-a:%{rake} plugin:apipie:cache[%%{gem_name}] FOREMAN_APIPIE_LANGS=en_US RAILS_ENV=production cache_part=resources OUT=%%{buildroot}%%{%{name}_apipie_cache_plugin} DATABASE_URL=nulldb://nohost --trace} \\
```
Comment (ekohl) [inline packages/foreman/foreman/foreman.spec:878]: Please check this carefully. I think this results in an explicit `/usr/bin/rake` in the generated macro

Comment (evgeni) [reply packages/foreman/foreman/foreman.spec:878]: I see in the generated file:

```
%{?-s:/usr/bin/rake %{-r*}%{!?-r:plugin:assets:precompile[%{-n*}%{!?-n:%{gem_name}}]} RAILS_ENV=production DATABASE_URL=nulldb://nohost --trace} \
%{?-a:/usr/bin/rake plugin:apipie:cache[%{gem_name}] FOREMAN_APIPIE_LANGS=en_US RAILS_ENV=production cache_part=resources OUT=%{buildroot}%{foreman_apipie_cache_plugin} DATABASE_URL=nulldb://nohost --trace} \
```

which looks correct to me?

**Review by evgeni — APPROVED** [review]

File: packages/foreman/foreman/foreman.spec (hunk 2 above)
Comment (ekohl) [reply packages/foreman/foreman/foreman.spec:878]: It does, but macros in macros is always tricky. Always good to be careful around that.

**Pull request merged**